Source/Core/loadKTX.js

```diff
@@ -0,0 +1,198 @@
+        var pixelDepth = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var numberOfArrayElements = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var numberOfFaces = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var numberOfMipmapLevels = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        var bytesOfKeyValueByteSize = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+
+        // TODO: read metadata? At least need to check for KTXorientation
+        byteOffset += bytesOfKeyValueByteSize;
+
+        var imageSize = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+
+        var texture = new Uint8Array(data, byteOffset, imageSize);
+
+        // Some tools use a sized internal format.
+        // See table 2: https://www.opengl.org/sdk/docs/man/html/glTexImage2D.xhtml
+        if (glInternalFormat === 0x8051) {         // GL_RGB8
+            glInternalFormat = PixelFormat.RGB;
+        } else if (glInternalFormat === 0x8058) {  // GL_RGBA8
+            glInternalFormat = PixelFormat.RGBA;
+        }
+
+        if (!PixelFormat.validate(glInternalFormat)) {
+            throw new RuntimeError('glInternalFormat is not a valid format.');
+        }
+
+        if (PixelFormat.isCompressedFormat(glInternalFormat)) {
+            if (glType !== 0) {
+                throw new RuntimeError('glType must be zero when the texture is compressed.');
+            }
+            if (glTypeSize !== 1) {
+                throw new RuntimeError('The type size for compressed textures must be 1.');
+            }
+            if (glFormat !== 0) {
+                throw new RuntimeError('glFormat must be zero when the texture is compressed.');
+            }
+            if (numberOfMipmapLevels === 0) {
+                throw new RuntimeError('Generating mipmaps for a compressed texture is unsupported.');
+            }
+        } else {
+            if (glBaseInternalFormat !== glFormat) {
+                throw new RuntimeError('The base internal format must be the same as the format for uncompressed textures.');
+            }
+        }
+
+        if (pixelDepth !== 0) {
+            throw new RuntimeError('3D textures are unsupported.');
+        }
+
+        // TODO: support texture arrays and cubemaps
+        if (numberOfArrayElements !== 0) {
+            throw new RuntimeError('Texture arrays are unsupported.');
+        }
+        if (numberOfFaces !== 1) {
+            throw new RuntimeError('Cubemaps are unsupported.');
+        }
+
+        // TODO: multiple mipmap levels
+        if (PixelFormat.isCompressedFormat(glInternalFormat) && numberOfMipmapLevels > 1) {
+            var levelSize = PixelFormat.compressedTextureSize(glInternalFormat, pixelWidth, pixelHeight);
+            texture = new Uint8Array(texture.buffer, 0, levelSize);
+        }
+
+        return {
```

Comment:
Can we document this type?  Even if it is a trivial constructor + readonly properties?

I don't know if there is an easier way to document this (other than making the type) like how we can document callback functions.